bokeh/_glyph_functions.py

```diff
@@ -52,14 +59,19 @@ def func(plot, *args, **kwargs):
 
         _update_plot_data_ranges(plot, datasource, x_data_fields, y_data_fields)
         kwargs.update(glyph_params)
-
-        glyph_props = glyphclass.properties() | set(argnames)
-        glyph_kwargs = dict((key, value) for (key, value) in iteritems(kwargs) if key in glyph_props)
-        glyph = glyphclass(**glyph_kwargs)
-
+    
         nonselection_glyph_params = _materialize_colors_and_alpha(kwargs, prefix='nonselection_', default_alpha=0.1)
+        
+        #glyph_props = glyphclass.properties() | set(argnames)
+        #glyph_kwargs = dict((key, value) for (key, value) in iteritems(kwargs) if key in glyph_props)
+        #glyph = glyphclass(**glyph_kwargs)
```

Comment:
I agree. This is still WIP and was waiting for your blessing with this approach before really going forward with this :)